### PR TITLE
Feature: Styled Components-like API

### PR DESCRIPTION
**Pull request by gksander**
Addresses #19, adding a Styled Components-like API. [See here](https://formidable-com-react-native-zephyr-staging-20.surge.sh/open-source/react-native-zephyr/styling-methods#the-styled-component-wrapper) for the docs-changes preview, which gives a better idea of what this API looks like.

(Sorry for the gnarly code LOL)

### CODE REVIEW

File: packages/sample/AppBody.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import * as React from "react";
-import { StyledImage, StyledText, StyledView } from "./styled";
```
Comment (gksander) [inline packages/sample/AppBody.tsx:2]: This is for the sample app, changes are not meaningful. Just playing around.

File: packages/core/src/createStyleBuilder.test.ts [hunk 2]
```diff
@@ -1,101 +0,0 @@
-import { vi, describe, it, expect } from "vitest";
```
Comment (gksander) [inline packages/core/src/createStyleBuilder.test.ts:1]: File was renamed to a `.tsx` extension.

File: packages/core/src/createStyleBuilder.test.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          tiny: 0.5 * baseFontSize,
+        },
+      }),
+    });
+
+    expect(styles("m:tiny")).toEqual({ margin: 0.5 * 16 });
+  });
+});
+
+describe("createStyleBuilder().styled", () => {
```
Comment (gksander) [inline packages/core/src/createStyleBuilder.test.tsx:113]: This is the new section of this test file.

File: packages/core/src/createStyleBuilder.test.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  const { styled } = createStyleBuilder();
+  beforeEach(() => {
+    colorScheme = "light";
+  });
+
+  it("wraps a component and adds style.", () => {
+    const MyText = styled(Text)("color:red-100");
+    render(<MyText>Hey world</MyText>);
+
+    // @ts-expect-error HALP. How do I type this mock?
```
Comment (gksander) [inline packages/core/src/createStyleBuilder.test.tsx:123]: I'll probably file this as a cleanup ticket to try to come back and get the types wrangled here.

**Review by kadikraman — APPROVED** [review]
Nice! I went through the whole PR several times including running it locally, and I can't find anything to complain about - LGTM!

**Pull request merged**